### ISSUE TITLE
Notebook toolbar buttons do not respect context

### ISSUE THREAD

**Issue by agoose77** [jupyterlab/jupyterlab#12177]
## Description

When clicking a button on the notebook toolbar, JupyterLab does not update the current document from the click event. This means that one can restart the kernel of second open notebook tab if it is the current active document, and one then directly clicks the first notebook toolbar.

## Reproduce
Clicking "Restart and Run All" on the right notebook acts on the left notebook kernel, if that notebook body was most recently interacted with.
![image](https://user-images.githubusercontent.com/1248413/157093495-bd808bf2-ccb1-4874-a534-91d081b59581.png)

## Context

- JupyterLab version: 3.3.0

**Comment by jtpio** [comment]
Thanks @agoose77 for reporting.

Does this happen only with "Restart and Run All", or also with other buttons?

**Comment by Maxsparrow** [comment]
I also ran into this issue, so I did some testing.

The buttons below do not switch context at all when pushed - they run on the selected notebook (selected meaning the one last interacted with that has the blue bar on top), not the notebook with the corresponding button, and the selected notebook doesn't change:
* Restart and run all
* Restart Kernel

An example of this first issue. Left notebook is selected, but I push restart on the right notebook:
![Screen Shot 2022-04-01 at 8 51 04 AM](https://user-images.githubusercontent.com/7087373/161277333-3d4a7382-d5cf-4120-9f61-c4f93c391483.png)

Left notebook is restarted, and left notebook remains selected:
![image](https://user-images.githubusercontent.com/7087373/161277268-5bc8150f-0dfe-4177-a984-d9f0a9d3d45c.png)

This is slightly different - these buttons effect the selected notebook, and _after_ that it selects the notebook corresponding to the button where they were pressed:
* Stop kernel
* Cut
* Copy
* Paste
* Add Cell
* Save


An example of the second issue. Right notebook is selected, but I am pushing the stop kernel button on the left notebook:
![Screen Shot 2022-04-01 at 8 53 36 AM](https://user-images.githubusercontent.com/7087373/161277689-2b9d851b-e82a-4e19-85df-ef5fab0a5a33.png)

Kernel stops on right notebook, then left notebook becomes selected:
![image](https://user-images.githubusercontent.com/7087373/161277658-8c9c97c2-2742-40a2-a754-c87c42cc0e33.png)


Kernel selection and cell type selector work as expected.

Tested in a fresh JupyterLab install, version 3.3.2, no extensions.
